R/seroprevalence_data.R

```diff
@@ -139,6 +139,32 @@ prepare_bin_data <- function(serodata) {
   return(xx)
 }
 
+#' Function that returns the probability of being seropositive for age-varying
+#' FoI including seroreversion
+#'
+#' @param exposure_age Integer corresponding to the age of the exposed cohort
+#' @param foi Numeric atomic vector corresponding to the age-varying FoI to
+#' simulate from
+#' @param mu Seroreversion rate
+#' @return probability of being seropositive for age-varying FoI
+#' including seroreversion
+get_probability_exact_av <- function(
+    exposure_age,
+    foi,
+    mu = 0
+) {
+  # TODO: Avoid for loop if possible
+  probability <- 0
+  # solves ODE exactly within pieces
+  for (i in 1:exposure_age) {
+    probability <-
+      (1 / (foi[i] + mu)) * exp(- (foi[i] + mu)) *
+      (foi[i] * (exp(foi[i] + mu)  - 1) + probability * (foi[i] + mu)
+      )
+  }
+  return(probability)
+}
+
 #' Function that generates the probabilities of being previously exposed to a
 #' pathogen given a historical Force-of-Infection.
```

Comment:
lowercase "force-of-infection" since it's not a name

Reply:
Also, no need to start off function with "Function" here

Reply:
Same as above (see #164).

Reply:
Cool, makes sense